applications/crossbar/src/modules/cb_allotments.erl

```diff
@@ -0,0 +1,326 @@
+-spec allowed_methods() -> http_methods().
+allowed_methods() ->
+    [?HTTP_GET, ?HTTP_POST].
+
+allowed_methods(?CONSUMED) ->
+    [?HTTP_GET].
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% This function determines if the provided list of Nouns are valid.
+%%
+%% Failure here returns 404
+%% @end
+%%--------------------------------------------------------------------
+-spec resource_exists() -> 'true'.
+-spec resource_exists(path_token()) -> 'true'.
+resource_exists() -> 'true'.
+resource_exists(?CONSUMED) -> 'true'.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% This function determines if the parameters and content are correct
+%% for this request
+%%
+%% Failure here returns 400
+%% @end
+%%--------------------------------------------------------------------
+-spec validate(cb_context:context()) -> cb_context:context().
+validate(Context) ->
+    validate_allotments(Context, cb_context:req_verb(Context)).
+
+-spec validate(cb_context:context(), path_token()) -> cb_context:context().
+validate(Context, ?CONSUMED) ->
+    validate_consumed(Context, cb_context:req_verb(Context)).
+
+-spec validate_allotments(cb_context:context(), http_method()) -> cb_context:context().
+validate_allotments(Context, ?HTTP_GET) ->
+    load_allotments(Context);
+validate_allotments(Context, ?HTTP_POST) ->
+    cb_context:validate_request_data(<<"allotments">>, Context, fun on_successful_validation/1).
+
+-spec validate_consumed(cb_context:context(), http_method()) -> cb_context:context().
+validate_consumed(Context, ?HTTP_GET) ->
+    load_consumed(Context).
+
+-spec post(cb_context:context()) -> cb_context:context().
+post(Context) ->
+    update_allotments(Context).
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Load a document from the database
+%% @end
+%%--------------------------------------------------------------------
+-spec load_allotments(cb_context:context()) -> cb_context:context().
+load_allotments(Context) ->
+    Context1 = maybe_handle_load_failure(crossbar_doc:load(?PVT_TYPE, Context)),
+    Allotments = wh_json:get_json_value(?PVT_ALLOTMENTS, cb_context:doc(Context1), wh_json:new()),
+    cb_context:set_resp_data(Context1, Allotments).
+
+
+-spec load_consumed(cb_context:context()) -> cb_context:context().
+load_consumed(Context) ->
+    Context1 = crossbar_doc:load(?PVT_TYPE, Context),
+    Allotments = wh_json:get_json_value(?PVT_ALLOTMENTS, cb_context:doc(Context1), wh_json:new()),
+    load_consumed(Context, Allotments).
+
+-spec load_consumed(cb_context:context(), wh_json:objects()) -> cb_context:context().
+load_consumed(Context, Allotments) ->
+    ReqCycleStart = cb_context:req_param(Context, <<"start">>),
```

Comment:
I'd like to see `from` and `to` used, to mirror `created_from/created_to` and `modified_from/modified_to`. Even better, you could use `consumed_from/consumed_to` and look at adapting to use `cb_modules_util:range_view_options` and `cb_modules_util:range_modb_view_options` to get the list of MODbs needed to fulfill the valid range specified.